subprojects/core/src/main/java/org/gradle/execution/plan/DefaultNodeValidator.java

```diff
@@ -75,7 +75,7 @@ private void logWarnings(List<? extends InternalProblem> problems) {
                 String warning = convertToSingleLine(renderMinimalInformationAbout(problem, false, false));
                 withDocumentation(problem, DeprecationLogger.deprecateBehaviour(warning)
                     .withContext("Execution optimizations are disabled to ensure correctness.")
-                    .willBeRemovedInGradle9())
+                    .willBecomeAnErrorInNextMajorGradleVersion())
```

Comment:
This works, and I had a similar impulse to use this variant. However, being explicit may be best. If we forget to address it before Gradle 10, it will silently start warning that this will become an error in Gradle 11 instead. Also, any usages of `willBecomeAnErrorInNextMajorGradleVersion` during 9.x that are forgotten will be undistinguishable from those added during Gradle 10.x.

Reply:
+1
I am not sure we should have this method at all